File: Csharp/Show/Show/ConcreteDecoratorA.cs
namespace Show
{
     class ConcreteDecoratorA: Decorator
    {
        public ConcreteDecoratorA(Component component) : base(component)
        {
        }
        public override string Operation()
        {
            return $"ConcreteDecoratorA({base.Operation()})";
        }


    }
}

File: Language/JavaScript/Structural/Decorator/Example1/ConcreteDecoratorA.js
// ConcreteDecoratorA.js

const Decorator = require('./Decorator');

// ConcreteDecoratorA adds behavior to the wrapped component
class ConcreteDecoratorA extends Decorator {
    operation() {
        // add extra behavior
        return `ConcreteDecoratorA(${super.operation()})`;
    }
}

module.exports = ConcreteDecoratorA;